modules/apps/notification/notification-rest-api/src/main/java/com/liferay/notification/rest/dto/v1_0/util/NotificationUtil.java

```diff
@@ -37,18 +46,46 @@
 public class NotificationUtil {
 
 	public static NotificationContext toNotificationContext(
-			com.liferay.notification.rest.dto.v1_0.NotificationTemplate
-				notificationTemplate)
-		throws Exception {
+		com.liferay.notification.rest.dto.v1_0.NotificationTemplate
+			notificationTemplate,
+		ObjectFieldLocalService objectFieldLocalService) {
 
 		NotificationContext notificationContext = new NotificationContext();
 
-		notificationContext.setAttachmentObjectFieldIds(
-			ListUtil.fromArray(
-				notificationTemplate.getAttachmentObjectFieldIds()));
-
 		notificationContext.setType(notificationTemplate.getType());
 
+		List<Long> attachmentObjectFieldIds = new ArrayList<>();
+
+		if (GetterUtil.getBoolean(PropsUtil.get("feature.flag.LPS-168220"))) {
+			for (String attachmentObjectFieldERC :
+					ListUtil.fromArray(
+						notificationTemplate.getAttachmentObjectFieldERCs())) {
+
+				ObjectField objectField =
+					objectFieldLocalService.fetchObjectField(
+						attachmentObjectFieldERC,
+						notificationTemplate.getObjectDefinitionId());
+
+				if (objectField == null) {
+					attachmentObjectFieldIds.clear();
+
+					break;
+				}
+
+				attachmentObjectFieldIds.add(objectField.getObjectFieldId());
+			}
+		}
+
+		if (attachmentObjectFieldIds.isEmpty()) {
+			notificationContext.setAttachmentObjectFieldIds(
+				ListUtil.fromArray(
+					notificationTemplate.getAttachmentObjectFieldIds()));
+		}
+		else {
+			notificationContext.setAttachmentObjectFieldIds(
+				attachmentObjectFieldIds);
```

Comment:
@marcelabc I would keep the the alphabetical order, i.e., `setAttachmentObjectFieldIds` before `setType`